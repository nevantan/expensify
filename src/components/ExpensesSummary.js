import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
  <div>
    <p>
      Viewing { props.expenseCount }
      { props.expenseCount === 1 ? ' expense ' : ' expenses ' }
      totalling { numeral(props.expensesTotal / 100).format('$0,0.00') }
    </p>
  </div>
);

const mapStateToProps = (state) => {
  const filtered = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: filtered.length,
    expensesTotal: selectExpensesTotal(filtered)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);