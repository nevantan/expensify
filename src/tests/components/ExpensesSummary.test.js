import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should properly render summary for single expense', () => {
  let wrapper = shallow(
    <ExpensesSummary
      expenseCount={1}
      expensesTotal={9434}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should properly render summary for multiple expenses', () => {
  let wrapper = shallow(
    <ExpensesSummary
      expenseCount={2}
      expensesTotal={9434}
    />
  );
  expect(wrapper).toMatchSnapshot();
});