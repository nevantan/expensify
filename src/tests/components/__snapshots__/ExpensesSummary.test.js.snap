// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render summary for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses 
    totalling 
    $94.34
  </p>
</div>
`;

exports[`should properly render summary for single expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense 
    totalling 
    $94.34
  </p>
</div>
`;
